packages/grant-explorer/src/features/round/ViewCartPage/SummaryContainer.tsx

```diff
@@ -275,13 +280,78 @@ export function SummaryContainer() {
     }
   }, [totalDonationAcrossChainsInUSDData]);
 
+  /* Matching estimates are calculated per-round */
+  const matchingEstimateParamsPerRound =
+    rounds?.map((round) => {
+      const projs = projects.filter((project) => project.roundId === round.id);
+      return {
+        roundId: getAddress(round.id ?? ""),
+        chainid: projs[0].chainId,
+        potentialVotes: projects.map((proj) => ({
+          amount: parseUnits(
+            proj.amount ?? "0",
+            getVotingTokenForChain(Number(proj.chainId) as ChainId).decimal ??
+              18
+          ),
+          recipient: proj.recipient,
+          contributor: address as Address,
+          token: getVotingTokenForChain(
+            Number(proj.chainId) as ChainId
+          ).address.toLowerCase(),
+        })),
+      };
+    }) ?? [];
+
+  const {
+    data: matchingEstimates,
+    error: matchingEstimateError,
+    isLoading: matchingEstimateLoading,
+  } = useMatchingEstimates(matchingEstimateParamsPerRound);
+
+  const estimateText = matchingEstimates
+    ?.flat()
+    .map((est) => est.differenceInUSD ?? 0)
+    .filter((diff) => diff > 0)
+    .reduce((acc, b) => acc + b, 0)
+    .toFixed(2);
+
```

Comment:
Looks like the same code from elsewhere, worth putting it into a function?

Reply:
Done.